testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java

```diff
@@ -669,12 +669,27 @@ private Optional<DistributedQueryRunner> tryGetDistributedQueryRunner()
         return Optional.empty();
     }
 
-    protected TestTable newTrinoTable(String namePrefix, @Language("SQL") String tableDefinition)
+    protected final TestTable newTrinoTable(String namePrefix, @Language("SQL") String tableDefinition)
     {
-        return newTrinoTable(namePrefix, tableDefinition, ImmutableList.of());
+        return newTrinoTable(namePrefix, tableDefinition, ImmutableList.of(), Optional.empty());
     }
 
-    protected TestTable newTrinoTable(String namePrefix, @Language("SQL") String tableDefinition, List<String> rowsToInsert)
+    protected final TestTable newTrinoTable(String namePrefix, @Language("SQL") String tableDefinition, String primaryKey)
+    {
+        return newTrinoTable(namePrefix, tableDefinition, ImmutableList.of(), Optional.of(primaryKey));
+    }
+
+    protected final TestTable newTrinoTable(String namePrefix, @Language("SQL") String tableDefinition, List<String> rowsToInsert)
+    {
+        return newTrinoTable(namePrefix, tableDefinition, rowsToInsert, Optional.empty());
+    }
+
+    protected final TestTable newTrinoTable(String namePrefix, @Language("SQL") String tableDefinition, List<String> rowsToInsert, String primaryKey)
```

Comment:
IMO the primary-key variants should be dedicated methods with names making clear that they are to be used with MERGE-related tests.

I suggest `newTrinoTableForMerge`?